client/derive_config.ts

```diff
@@ -17,7 +17,12 @@ export function deriveConfig(conf: ClientConfig = {}): Doc {
     const got: Doc = get({ profile: _conf.profile || "default" });
 
     if (typeof _conf.credentials !== "function") {
-      _conf.credentials = { ...got.credentials, ..._conf.credentials };
+      _conf.credentials = {
+        ...got.credentials,
```

Comment:
Should we purposely strip the accessKeyId and secretAccessKey off of this before spreading? It's polluting the object right now.

Reply:
as you correctly said b4 `got` just doesn't have `credentials`

Reply:
👍